src/main/java/at/hannibal2/skyhanni/features/misc/AbiphoneContactSlots.kt

```diff
@@ -0,0 +1,44 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiInventory
+import net.minecraftforge.event.entity.player.ItemTooltipEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class AbiphoneContactSlots {
+    private val ALL_UPGRADES_TOTAL: Int = 40
+    private val abiphoneContactSlotsGroup = RepoPattern.group("abiphone.contact.slots")
+    private val isAbiphoneInternalNamePattern by abiphoneContactSlotsGroup.pattern(("isabiphone.internalname"), ("ABIPHONE_.*"))
+    private val maximumContactSlotsLoreLinePattern by abiphoneContactSlotsGroup.pattern(("maximumcontactslots.loreline"), (" ?§7Maximum Contacts: (?:§.)*(?<first>\\d+)(?: |§.)*(?:[(\\[]\\+?(?<second>\\d+)[)\\]])?(?: |§.)*(?:[(\\[]\\+?(?<third>\\d+)[)\\]])?(?: |§.)*(?:[(\\[]\\+?(?<fourth>\\d+)[)\\]])?(?: |§.)*(?:[(\\[]\\+?(?<fifth>\\d+)[)\\]])?")) // adapted from nea's https://regex101.com/r/aCVGHN/1 -ery
+
+    @SubscribeEvent
+    fun onTooltip(event: ItemTooltipEvent) {
+        if (!(LorenzUtils.inSkyBlock && SkyHanniMod.feature.misc.abiphoneContactsProgress)) return
+        if (Minecraft.getMinecraft().currentScreen !is GuiInventory) return
+        val itemStack = event.itemStack
+        if (!(isAbiphoneInternalNamePattern.matches(itemStack.getInternalName().asString()))) return
```

Comment:
Please use the new ItemCategory instead
```suggestion
        if (itemStack.getItemCategoryOrNull() != ItemCategory.ABIPHONE) return
```